### PR TITLE
docs: fixed URLs for tutorial as described in #3241

### LINKED ISSUE #3241: [DOCS] Fix URLs in weak supervision with sentence transformers tutorial
## Which page or section is this issue related to?

In a couple different places (linked below), it references an earlier/previous weak supervision tutorial but the link leads you to the same page. 
- https://docs.argilla.io/en/latest/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.html#:~:text=Weak%20supervision%20with%20Argilla%20tutorial
- https://docs.argilla.io/en/latest/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.html#appendix:-visualize-changes:~:text=as%20in%20the-,previous%20tutorial,-%2C%20but%20on%20the
- https://docs.argilla.io/en/latest/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.html#appendix:-visualize-changes:~:text=model%20from%20the-,previous%20tutorial,-.%20Remember%20we%20achieved

Under the "Detailed Workflow" section, the following hyperlink does not exist at all.
- https://docs.argilla.io/en/latest/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.html#:~:text=weak%20supervision%20guide

References to the Appendix (which is at the bottom of the page) in step 5 do not work because the URLs don't match the capitalization for the header permalinks.

### PR DESCRIPTION
**Pull request by garimau**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
I fixed the hyperlinked URLs in the tutorial, so users would be referred to the right pages for other relevant tutorials.

Closes #3241 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)
- [X] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

N.A.

**Checklist**

- [X] I have merged the original branch into my forked branch
- [X] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb [hunk 1]
```diff
@@ -203,10 +210,11 @@
     "\n",
     "5. Once you are satisfied with your extended weak labels, use the extended matrix of the `WeakLabels` instance with your library/method of choice to build a training set or even train a downstream text classification model. You can iterate between this step and step 4 to try several thresholds and embeddings possibilities until you achieve a satisfactory result.\n",
     "\n",
-    "This guide shows you an end-to-end example using Snorkel. You could alternatively use any other label model available in Argilla. If you are interested in learning about other options, please check our [weak supervision guide](../../guides/techniques/weak_supervision.html)."
+    "This guide shows you an end-to-end example using Snorkel. You could alternatively use any other label model available in Argilla. If you are interested in learning about other options, please check our [weak supervision guide](https://docs.argilla.io/en/latest/guides/programmatic_labeling_with_rules.html)."
```
Comment (davidberenstein1957) [inline docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb]: Hi @garimau, we should make this relative URLs to the local files in the website. Something like this `/guides/programmatic_labeling_with_rules` should work. Could you check this with the `sphinx-autobuild` thingy?

File: docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb [hunk 2]
```diff
@@ -1651,9 +1679,9 @@
    "source": [
     "## 6. Training a downstream model\n",
     "\n",
-    "Now we will train the same dowstream model as in the [previous tutorial](./labelling-textclassification-sentencetransformers-weaksupervision.ipynb), but on the data produced by a label model from our extended weak labels.\n",
+    "Now we will train the same dowstream model as in the [previous tutorial](https://docs.argilla.io/en/latest/guides/programmatic_labeling_with_rules.html), but on the data produced by a label model from our extended weak labels.\n",
```
Comment (davidberenstein1957) [inline docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb]: same here about the local references

File: docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb [hunk 3]
```diff
@@ -1712,13 +1740,14 @@
...
   {
+   "attachments": {},
    "cell_type": "markdown",
    "id": "97eb8ed4-bf0c-4212-829e-0e96ef131efb",
    "metadata": {
     "id": "97eb8ed4-bf0c-4212-829e-0e96ef131efb"
    },
    "source": [
-    "Now let's see how our downstream model compares with the original model from the [previous tutorial](./labelling-textclassification-sentencetransformers-weaksupervision.ipynb).\n",
+    "Now let's see how our downstream model compares with the original model from the [previous tutorial](https://docs.argilla.io/en/latest/guides/programmatic_labeling_with_rules.html).\n",
```
Comment (davidberenstein1957) [inline docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb]: local references to files

File: docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb [hunk 4]
```diff
@@ -11,14 +12,15 @@
...
+   "attachments": {},
    "cell_type": "markdown",
    "id": "00335193",
    "metadata": {
     "id": "00335193"
    },
    "source": [
     "In this tutorial, we show how weak supervision workflows in Argilla can be extended with sentence embeddings.\n",
-    "We start from the weak supervision workflow presented in our [Weak supervision with Argilla tutorial](./labelling-textclassification-sentencetransformers-weaksupervision.ipynb) and improve on its results by extending the coverage of its rules.\n",
+    "We start from the weak supervision workflow presented in our [Weak supervision with Argilla tutorial](https://docs.argilla.io/en/latest/guides/programmatic_labeling_with_rules.html) and improve on its results by extending the coverage of its rules.\n",
```
Comment (davidberenstein1957) [inline docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb]: local references to files, see comment below

**Review by davidberenstein1957 — COMMENTED** [review]
Looks Good To Me (LGTM). Some minor remarks:)

File: docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb (hunk 4 above)
Comment (garimau) [reply docs/_source/tutorials/notebooks/labelling-textclassification-sentencetransformers-weaksupervision.ipynb]: they should be fixed now!

**Review by davidberenstein1957 — APPROVED** [review]

**Pull request merged**